dapps/marketplace/src/hoc/withFingerprint.js

```diff
@@ -24,7 +24,22 @@ function fetchFingerprint() {
     components
       .filter(x => browserPropsWhitelist.includes(x.key))
       .forEach(x => (browserProps[x.key] = x.value))
-    cachedFingerprintData = { fingerprint: hash, ...browserProps }
+
+    // Fingerprint injected by mobile app
+    let mobileFingerprint
+    try {
+      if (window.localStorage.deviceFingerprint) {
+        mobileFingerprint = JSON.parse(window.localStorage.deviceFingerprint)
```

Comment:
Not completely sure how hash is used, but might be useful to rehash the hash with additional `deviceFingerprint` data points. @franckc might know more here

Reply:
On mobile we already have the UUID so I don't think we need to hash the deviceFingerprint. I would leave as is.

Reply:
Thanks @sparrowDom and @franckc